### DISCUSSION TITLE
Allow `nullptr` when logging `ToSinkAlso()`

### DISCUSSION THREAD

**Discussion by jiawen** [abseil/abseil-cpp#1609] [Ideas]
Our project has a debugging mechanism whereby based on a config option, we may want to also log to an additional, independent `absl::LogSink*`. However `ToSinkAlso()` does not accept `nullptr` as a parameter - this is reasonable but forces us to use somewhat unsightly workarounds:

Either:
```
void MyFunc(absl::LogSink* optional_sink) {
if (optional_sink) {
  LOG(INFO).ToSinkAlso(optional_sink) << "my message";
} else {
  LOG(INFO) << "my message";
}
```

or:

define a `class NoopLogSink : public absl::LogSink` whose `Send()` does nothing and has a global singleton. Then the code looks like:

```
void MyFunc(absl::LogSink* optional_sink = NoopLogSink::singleton()) {
  LOG(INFO).ToSinkAlso(optional_sink) << "my message";
```

Is there a less ugly alternative that I'm missing? The only thing I can think of is to write a wrapper macro around all the `LOG` ones but it's a pretty big surface.

**Comment by derekmauro** [comment]
It might be possible to put the logic to decide whether to log directly into the sink that is currently optional.
